src/test/java/com/karankumar/bookproject/ui/components/dialog/ResetShelvesDialogTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+    The book project lets a user keep track of different books they've read, are currently reading or would like to read
+    Copyright (C) 2020  Karan Kumar
+
+    This program is free software: you can redistribute it and/or modify it under the terms of the
+    GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along with this program.
+    If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.karankumar.bookproject.ui.components.dialog;
+
+import com.github.mvysny.kaributesting.v10.MockVaadin;
+import com.github.mvysny.kaributesting.v10.Routes;
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.service.BookService;
+import com.karankumar.bookproject.ui.MockSpringServlet;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.spring.SpringServlet;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+
+@IntegrationTest
+public class ResetShelvesDialogTest {
+    private BookService bookService;
+
+    @Autowired private ApplicationContext ctx;
+    private static Routes routes;
+
+    @BeforeAll
+    public static void discoverRoutes() {
+        routes = new Routes().autoDiscoverViews("com.karankumar.bookproject.ui");
+    }
+
+    @BeforeEach
+    public void setup(@Autowired BookService bookService) {
+        this.bookService = bookService;
+        final SpringServlet servlet = new MockSpringServlet(routes, ctx);
+        MockVaadin.setup(UI::new, servlet);
+
+        Assumptions.assumeTrue(bookService != null);
+    }
+
+    @Test
+    void openDialog(){
```

Comment:
- A new test method is needed The test method name should be descriptive (it should describe what is being tested here) 
- A space is needed before the closing brace
- The problem with this test method is that assumes that the book repository already has more than one book, but this assumption is implicit -- it is not conveyed to the reader. It would be better if you added a populate method and invoked it at the start of this test. That method can add at least two books. Then, we can test whether all books were successfully deleted

Reply:
"A new test method is needed The test method name should be descriptive"
-- Okay, I've changed the method name to `openResetShelvesDialog`

"A space is needed before the closing brace"
-- Done

"The problem with this test method is that assumes that the book repository already has more than one book, but this assumption is implicit -- it is not conveyed to the reader. It would be better if you added a populate method and invoked it at the start of this test. That method can add at least two books. Then, we can test whether all books were successfully deleted"
-- Okay, Ive added the following method in the ResetShelvesDialogTest class:

`private void populateBookService(){
        Author h_a_Rey = new Author("H.A.", "Rey");
        PredefinedShelf shelf = new PredefinedShelf(PredefinedShelf.ShelfName.TO_READ);
        Book book = new Book("Curious George Takes A Job", h_a_Rey, shelf);
        bookService.save(book);
        book = new Book("Curious George Gets a medal", h_a_Rey, shelf);
        bookService.save(book);
    }`

I call this method at the end of the `setup()` method, right after `Assumptions.assumeTrue(bookService != null);`

I get an number of unwanted errors - obviously, Im arranging The predefined shelf wrong. any pointers?